eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -121,10 +122,88 @@ function Confirm-ChangeLogEntry {
   return $true
 }
 
-function Set-TestChangeLog($TestVersion, $changeLogFile, $ReleaseEntry) {
-  Set-Content -Path $changeLogFile -Value @"
-# Release History
-## $TestVersion ($(Get-Date -f "yyyy-MM-dd"))
-- $ReleaseEntry
-"@
+function New-ChangeLogEntry {
+  param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Version,
+    [String]$Status=$UNRELEASED_TAG,
+    [String]$Title,
+    [String[]]$Content,
+    [String]$IgnoreInvalids=$True
+  )
+
+  # Validate RelaseStatus
+  $Status = $Status.Trim().Trim("()")
+  if ($Status -ne "Unreleased") {
+    try {
+      $Status = ([DateTime]$Status).ToString($DATE_FORMAT)
+    }
+    catch {
+        LogWarning "Invalid date [ $Status ] passed as status for Version [$Version]. Please use a valid date in the format '$DATE_FORMAT' or use '$UNRELEASED_TAG'"
+        if (!$IgnoreInvalids) { exit 1 }
+    }
+  }
+  $Status = "($Status)"
+
+  # Validate Version
+  try {
+    if ($Language -eq "python") {
+      $Version = ([AzureEngSemanticVersion]::ParsePythonVersionString($Version)).ToString()
```

Comment:
Once you do the python detection in the constructor you shouldn't need to have this fork here.